Plugins/Public/munitioncntl/Main.cpp

```diff
@@ -0,0 +1,205 @@
+uint lastProcessedProjectile = 0;
+
+enum TRACKING_STATE {
+	TRACK_ALERT,
+	TRACK_NOALERT,
+	NOTRACK_NOALERT
+};
+
+void LoadSettings();
+
+BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
+{
+	srand((uint)time(0));
+	// If we're being loaded from the command line while FLHook is running then
+	// set_scCfgFile will not be empty so load the settings as FLHook only
+	// calls load settings on FLHook startup and .rehash.
+	if (fdwReason == DLL_PROCESS_ATTACH)
+	{
+		if (set_scCfgFile.length() > 0)
+			LoadSettings();
+	}
+	else if (fdwReason == DLL_PROCESS_DETACH)
+	{
+	}
+	return true;
+}
+
+/// Hook will call this function after calling a plugin function to see if we the
+/// processing to continue
+EXPORT PLUGIN_RETURNCODE Get_PluginReturnCode()
+{
+	return returncode;
+}
+
+bool bPluginEnabled = true;
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Loading Settings
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void LoadSettings()
+{
+	returncode = DEFAULT_RETURNCODE;
+
+	INI_Reader ini;
+
+	char szCurDir[MAX_PATH];
+	GetCurrentDirectory(sizeof(szCurDir), szCurDir);
+	string scPluginCfgFile = string(szCurDir) + "\\flhook_plugins\\munitioncntl.cfg";
+	if (ini.open(scPluginCfgFile.c_str(), false))
+	{
+		while (ini.read_header())
+		{
+			if (ini.is_header("NoTrackingAlert"))
+			{
+				while (ini.read_value())
+				{
+					if (ini.is_value("MissileArch"))
+					{
+						setNoTrackingAlertProjectiles.insert(CreateID(ini.get_value_string(0)));
+					}
+				}
+			}
+			else if (ini.is_header("TrackingBlacklist"))
+			{
+				uint missileArch = 0;
+				uint blacklistedTrackingTypesBitmap = 0;
+				while (ini.read_value())
+				{
+					if (ini.is_value("MissileArch"))
+					{
+						missileArch = CreateID(ini.get_value_string(0));
+					}
+					else if (ini.is_value("Type"))
+					{
+						string typeStr = ToLower(ini.get_value_string(0));
+						if (typeStr.find("fighter") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_FIGHTER;
+						if (typeStr.find("freighter") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_FREIGHTER;
+						if (typeStr.find("transport") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_TRANSPORT;
+						if (typeStr.find("gunboat") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_GUNBOAT;
+						if (typeStr.find("cruiser") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_CRUISER;
+						if (typeStr.find("capital") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_CAPITAL;
+						if (typeStr.find("guided") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_GUIDED;
+						if (typeStr.find("mine") != string::npos)
+							blacklistedTrackingTypesBitmap |= OBJ_MINE;
+					}
+				}
+				mapTrackingByObjTypeBlacklistBitmap[missileArch] = blacklistedTrackingTypesBitmap;
```

Comment:
Check that we got both values populated. If either is invalid console log and do not add.